### PR TITLE
modules: nordic: Allow separate nRF 802.15.4 radio driver build

### PR DESCRIPTION
**Pull request by rlubos**
So far, nRF 802.15.4 radio driver build was dependent on the 802.15.4
subsystem in Zephyr. While this was a reasonable approach for samples,
it prevented the radio driver from being built as a standalone entity,
which could be useful in some applications (e. g. running certain
tests).

Resolve this, by providing a separate set of Kconfigs for the radio
driver, therefore allowing to build it as a separate entity. The 802154
subsystem simply enables the radio driver module in this case.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

### CODE REVIEW

**Review by jukkar — APPROVED** [review]
LGTM

**Comment by tbursztyka** [comment]
Could you detail these "certain tests" a bit in your commit message?

**Comment by rlubos** [comment]
@tbursztyka It's our internal stuff, we have some radio driver tests which use the radio driver API directly (bare-metal), and we want to port them to run with Zephyr, still using the radio driver API though (at least for starters). That's why I need to be able to build the driver as a standalone entity (`ieee802154_nrf5.c` shim occupies some of the radio driver notification callbacks).

**Comment by rlubos** [comment]
@nashif Sorry about not adding DNM, completely slipped my mind. Thanks for catching this up!

File: modules/Kconfig.nordic [hunk 1]
```diff
@@ -3,6 +3,58 @@
 
 config HAS_NORDIC_DRIVERS
 	bool
+	default y if HAS_HW_NRF_RADIO_IEEE802154
+
+menu "Nordic drivers"
+	depends on HAS_NORDIC_DRIVERS
+
+menuconfig NRF_802154_RADIO_DRIVER
+	bool "Enable 802154 radio driver"
```
Comment (anangl) [inline modules/Kconfig.nordic]: I think you should rather use "IEEE 802.15.4" as in other prompts.

File: modules/Kconfig.nordic [hunk 2]
```diff
@@ -3,6 +3,58 @@
...
+	depends on HAS_HW_NRF_RADIO_IEEE802154
+
+if NRF_802154_RADIO_DRIVER
+
+choice NRF_802154_CCA_MODE
+	prompt "nRF52 IEEE 802.15.4 CCA mode"
```
Comment (anangl) [inline modules/Kconfig.nordic]: nRF53 also features IEEE 802.15.4 mode, so perhaps these prompts should be updated then. Though I'm not sure if it should be done in this PR.

Comment (rlubos) [reply modules/Kconfig.nordic]: I'll simply make it `nRF` then.

**Comment by rlubos** [comment]
@anangl I've cleaned up the descriptions a bit.

**Review by anangl — APPROVED** [review]

**Comment by rlubos** [comment]
I've updated the manifest file.

**Review by carlescufi — APPROVED** [review]

**Pull request merged**